jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -5354,4 +5354,45 @@ public IntStream codePoints() {
 		return value;
 	}
 /*[ENDIF] Java10 */	
+
+/*[IF Java11]*/
+	/**
+	 * Returns a string object containing the character (Unicode code point)
+	 * specified.
+	 * 
+	 * @param codePoint
+	 *            a Unicode code point.
+	 * @return a string containing the character (Unicode code point) supplied.
+	 * @throws IllegalArgumentException
+	 *             if the codePoint is not a valid Unicode code point.
+	 * @since 11
+	 */
+	static String valueOfCodePoint(int codePoint) {
+		String string;
+		// Save less often check (codePoint > Character.MAX_CODE_POINT) last
+		if (codePoint < Character.MIN_CODE_POINT) {
```

Comment:
I would vouch to say this is also an uncommon path and should be moved with the `else` path. Do we agree?

Reply:
We need to check that before checking `} else if (codePoint <= 255) {` or we need to move it into the check for `<= 255`.  Either way, its got to be first.

Reply:
Sure, but merging the two checks has the benefit of smaller footprint and less code to jump over to get to the `0 <= codePoint <= 255` case.

Reply:
I strongly dislike the duplication of the second form.  If we can't optimize the first form reasonably then we're going to have problems with most user code.

Reply:
It seems you are suggesting code like following:
```
if (codePoint >= Character.MIN_CODE_POINT && codePoint <= 255) {
  ...
} else if (codePoint >= Character.MIN_CODE_POINT && codePoint < Character.MIN_SUPPLEMENTARY_CODE_POINT) {
  ...
} else if (codePoint >= Character.MIN_CODE_POINT && codePoint <= Character.MAX_CODE_POINT) {
  ...
} else {
  error handling
}
```
This introduces extra code and confusion (later reader need lots of explanation about the rational of this code logic). I don't think any performance gain here worth the potential issue later. 
Now I feel this code should go back to original structure, i.e.,
```
if ((codePoint < Character.MIN_CODE_POINT) || (codePoint > Character.MAX_CODE_POINT)) {
  error handling
} else if (codePoint <= 255) {
  ...
} else if (codePoint < Character.MIN_SUPPLEMENTARY_CODE_POINT) {
  ...
} else {
  ...
}
```
It is clear and easy to understand. Can we modify it to improve performance later if this becomes a hot spot?

Reply:
Yes, I hadn't thought about the extra checks introduced in my original comment. I'm fine either keeping as is or reverting back to your latest suggestion @JasonFengJ9. Apologies for the churn!

Reply:
No problem, updated PR as latest suggestion above.